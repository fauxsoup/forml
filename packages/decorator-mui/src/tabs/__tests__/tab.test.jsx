import { Tab } from "../";
import Context from "@forml/context";
import React from "react";
import { render } from "@testing-library/react";
import * as decorator from "../";

describe("renders", function() {
    let form;
    let parent;
    let title = "title";
    let description = "description";

    beforeEach(function() {
        form = { type: 'fieldset', items: [{ key: [] }] };
        parent = { type: 'tabs', tabs: [form] };
    });

    describe("with form options", function() {
        let fields = {
            icon: ['person', undefined],
            elevation: [0, 1, undefined]
        };

        Object.keys(fields).forEach(function(field) {
            fields[field].forEach(function(value) {
                describe(`${field}`, function() {
                    test(`${value}`, function() {
                        form = { ...form, [field]: value };
                        const { container } = render(
                            <Context.Provider value={{ decorator }}>
                                <Tab
                                    form={form}
                                    parent={parent}
                                    title={title}
                                    description={description}
                                />
                            </Context.Provider>
                        );

                        expect(container).toMatchSnapshot();
                    });
                    test(`${value} with layout`, function() {
                        parent = { ...parent, layout: 'horizontal' }
                        form = { ...form, [field]: value };
                        const { container } = render(
                            <Context.Provider value={{ decorator }}>
                                <Tab
                                    form={form}
                                    parent={parent}
                                    title={title}
                                    description={description}
                                />
                            </Context.Provider>
                        );

                        expect(container).toMatchSnapshot();
                    });
                });
            });
        });
    });
});
