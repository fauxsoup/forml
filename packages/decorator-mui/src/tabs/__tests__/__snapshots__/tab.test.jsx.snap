// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with form options elevation 0 1`] = `
<div>
  <div
    aria-disabled="false"
    class="MuiButtonBase-root MuiListItem-root makeStyles-root-25 makeStyles-vertical-27 MuiListItem-gutters MuiListItem-divider MuiListItem-button"
    role="button"
    tabindex="0"
  >
    <div
      class="MuiListItemIcon-root"
    >
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        dynamic_form
      </span>
    </div>
    <div
      class="MuiListItemText-root MuiListItemText-multiline"
    >
      <span
        class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-noWrap MuiTypography-displayBlock"
      >
        title
      </span>
      <span
        class="MuiTypography-root MuiListItemText-secondary MuiTypography-caption MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        description
      </span>
    </div>
    <span
      class="MuiTouchRipple-root"
    />
  </div>
</div>
`;

exports[`renders with form options elevation 0 with layout 1`] = `
<div>
  <div
    aria-disabled="false"
    class="MuiButtonBase-root MuiListItem-root makeStyles-root-31 makeStyles-horizontal-35 MuiListItem-gutters MuiListItem-divider MuiListItem-button"
    role="button"
    tabindex="0"
  >
    <div
      class="MuiListItemIcon-root"
    >
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        dynamic_form
      </span>
    </div>
    <div
      class="MuiListItemText-root MuiListItemText-multiline"
    >
      <span
        class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-noWrap MuiTypography-displayBlock"
      >
        title
      </span>
      <span
        class="MuiTypography-root MuiListItemText-secondary MuiTypography-caption MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        description
      </span>
    </div>
    <span
      class="MuiTouchRipple-root"
    />
  </div>
</div>
`;

exports[`renders with form options elevation 1 1`] = `
<div>
  <div
    aria-disabled="false"
    class="MuiButtonBase-root MuiListItem-root makeStyles-root-37 makeStyles-vertical-39 MuiListItem-gutters MuiListItem-divider MuiListItem-button"
    role="button"
    tabindex="0"
  >
    <div
      class="MuiListItemIcon-root"
    >
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        dynamic_form
      </span>
    </div>
    <div
      class="MuiListItemText-root MuiListItemText-multiline"
    >
      <span
        class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-noWrap MuiTypography-displayBlock"
      >
        title
      </span>
      <span
        class="MuiTypography-root MuiListItemText-secondary MuiTypography-caption MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        description
      </span>
    </div>
    <span
      class="MuiTouchRipple-root"
    />
  </div>
</div>
`;

exports[`renders with form options elevation 1 with layout 1`] = `
<div>
  <div
    aria-disabled="false"
    class="MuiButtonBase-root MuiListItem-root makeStyles-root-43 makeStyles-horizontal-47 MuiListItem-gutters MuiListItem-divider MuiListItem-button"
    role="button"
    tabindex="0"
  >
    <div
      class="MuiListItemIcon-root"
    >
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        dynamic_form
      </span>
    </div>
    <div
      class="MuiListItemText-root MuiListItemText-multiline"
    >
      <span
        class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-noWrap MuiTypography-displayBlock"
      >
        title
      </span>
      <span
        class="MuiTypography-root MuiListItemText-secondary MuiTypography-caption MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        description
      </span>
    </div>
    <span
      class="MuiTouchRipple-root"
    />
  </div>
</div>
`;

exports[`renders with form options elevation undefined 1`] = `
<div>
  <div
    aria-disabled="false"
    class="MuiButtonBase-root MuiListItem-root makeStyles-root-49 makeStyles-vertical-51 MuiListItem-gutters MuiListItem-divider MuiListItem-button"
    role="button"
    tabindex="0"
  >
    <div
      class="MuiListItemIcon-root"
    >
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        dynamic_form
      </span>
    </div>
    <div
      class="MuiListItemText-root MuiListItemText-multiline"
    >
      <span
        class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-noWrap MuiTypography-displayBlock"
      >
        title
      </span>
      <span
        class="MuiTypography-root MuiListItemText-secondary MuiTypography-caption MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        description
      </span>
    </div>
    <span
      class="MuiTouchRipple-root"
    />
  </div>
</div>
`;

exports[`renders with form options elevation undefined with layout 1`] = `
<div>
  <div
    aria-disabled="false"
    class="MuiButtonBase-root MuiListItem-root makeStyles-root-55 makeStyles-horizontal-59 MuiListItem-gutters MuiListItem-divider MuiListItem-button"
    role="button"
    tabindex="0"
  >
    <div
      class="MuiListItemIcon-root"
    >
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        dynamic_form
      </span>
    </div>
    <div
      class="MuiListItemText-root MuiListItemText-multiline"
    >
      <span
        class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-noWrap MuiTypography-displayBlock"
      >
        title
      </span>
      <span
        class="MuiTypography-root MuiListItemText-secondary MuiTypography-caption MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        description
      </span>
    </div>
    <span
      class="MuiTouchRipple-root"
    />
  </div>
</div>
`;

exports[`renders with form options icon person 1`] = `
<div>
  <div
    aria-disabled="false"
    class="MuiButtonBase-root MuiListItem-root makeStyles-root-1 makeStyles-vertical-3 MuiListItem-gutters MuiListItem-divider MuiListItem-button"
    role="button"
    tabindex="0"
  >
    <div
      class="MuiListItemIcon-root"
    >
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        person
      </span>
    </div>
    <div
      class="MuiListItemText-root MuiListItemText-multiline"
    >
      <span
        class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-noWrap MuiTypography-displayBlock"
      >
        title
      </span>
      <span
        class="MuiTypography-root MuiListItemText-secondary MuiTypography-caption MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        description
      </span>
    </div>
    <span
      class="MuiTouchRipple-root"
    />
  </div>
</div>
`;

exports[`renders with form options icon person with layout 1`] = `
<div>
  <div
    aria-disabled="false"
    class="MuiButtonBase-root MuiListItem-root makeStyles-root-7 makeStyles-horizontal-11 MuiListItem-gutters MuiListItem-divider MuiListItem-button"
    role="button"
    tabindex="0"
  >
    <div
      class="MuiListItemIcon-root"
    >
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        person
      </span>
    </div>
    <div
      class="MuiListItemText-root MuiListItemText-multiline"
    >
      <span
        class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-noWrap MuiTypography-displayBlock"
      >
        title
      </span>
      <span
        class="MuiTypography-root MuiListItemText-secondary MuiTypography-caption MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        description
      </span>
    </div>
    <span
      class="MuiTouchRipple-root"
    />
  </div>
</div>
`;

exports[`renders with form options icon undefined 1`] = `
<div>
  <div
    aria-disabled="false"
    class="MuiButtonBase-root MuiListItem-root makeStyles-root-13 makeStyles-vertical-15 MuiListItem-gutters MuiListItem-divider MuiListItem-button"
    role="button"
    tabindex="0"
  >
    <div
      class="MuiListItemText-root MuiListItemText-multiline"
    >
      <span
        class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-noWrap MuiTypography-displayBlock"
      >
        title
      </span>
      <span
        class="MuiTypography-root MuiListItemText-secondary MuiTypography-caption MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        description
      </span>
    </div>
    <span
      class="MuiTouchRipple-root"
    />
  </div>
</div>
`;

exports[`renders with form options icon undefined with layout 1`] = `
<div>
  <div
    aria-disabled="false"
    class="MuiButtonBase-root MuiListItem-root makeStyles-root-19 makeStyles-horizontal-23 MuiListItem-gutters MuiListItem-divider MuiListItem-button"
    role="button"
    tabindex="0"
  >
    <div
      class="MuiListItemText-root MuiListItemText-multiline"
    >
      <span
        class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-noWrap MuiTypography-displayBlock"
      >
        title
      </span>
      <span
        class="MuiTypography-root MuiListItemText-secondary MuiTypography-caption MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        description
      </span>
    </div>
    <span
      class="MuiTouchRipple-root"
    />
  </div>
</div>
`;
